### DISCUSSION TITLE
Getting the robot's current position using the API?

### DISCUSSION THREAD

**Discussion by theorician** [Hypfer/Valetudo#1749] [Q&A / Support]
I'm sure someone's answered this before but I can't find anything. I've checked Swagger and don't see anything obvious. What endpoint do I use to find the robot's current x,y coordinates?

Thanks!

**Comment by theorician** [comment]
I assume that I can get it via /v2/robot/state or /v2/robot/state/map but I was hoping there was something a bit more lightweight than that.

**Reply by Hypfer** [reply]
Because the robot's firmware can rotate, shift or otherwise modify the map at any time, such an endpoint would not make sense as coordinates alone become just a pair of meaningless numbers without their context which is the map data they're contained in.

**Reply by theorician** [reply]
Ah. Yeah, okay, that makes perfect sense. Thank you for the quick reply.

Here's what I do - which is probably something you're going to hate. 

The maps are persistent so the coordinates are persistent too (at least in my experience and with my robot.) 

1. I have a home assistant automation which sends the robot to a set of coordinates to make it easier to empty it. 
2. I also have a predefined list of no-go-areas (which is where the carpets are) as a json snippet in HA, and when I want my robot to mop, an automation sets up the no-go-areas. (My robot does not have a no-mop-zone feature.) When it's done mopping, it clears the no-go-areas again so that it can vacuum the carpets as normal in subsequent runs.

Both these rely on me inserting coordinates into HA automations which isn't great of course. I've not found a better way to do it though. Am I just overlooking some obvious things?

**Reply by Hypfer** [reply]
> which is probably something you're going to hate.

Eh. I just hate it when people try force things into the project.
If it works in your setup and you're aware of the limitations then that's fine. Feel free to build something around that.
It just can't be added to the API because it's based on assumptions that will not always be true

> I've not found a better way to do it

The obvious response to this is "buy a robot that has support for no-mop-zones".
That's the solution to your problem.

If it can't do X, buy something that can do X.
Do not try to bend reality so that a shallow image of X may be achieved sometimes under some circumstances.

This is especially relevant here, because every properly mopping supported robot has no-mop-zones.
The only robots that don't have support for no-mop-zones feature "mopping" that is borderline useless.

You're investing a lot of time into hacking something together that is a very very very subpar experience.
Not that I'm stopping you of course. If it's a fun project then sure, do whatever you like.

**Reply by theorician** [reply]
That's a fair point. I was looking at upgrading to the S7. I have the S5 at the moment. I think it doesn't have good no-mop-zone support, but I might just be doing it wrong.

Additionally to my hacked-together no-mop-zone support, what this also lets me do is 'quick clean' support, where it adds known-problematic areas of the house (such as curtains or some chairs) that it has problems with. Ordinarily I'd move those out of the way, but sometimes I just want it to do a run without me having to prep the house first.
